### DISCUSSION TITLE
FTP-Deploy-Action with a specific brancha

### DISCUSSION THREAD

**Discussion by ntn888** [community/community#25306] [Actions]
<p>I have this deply workflow:</p>
<pre><code class="lang-auto">jobs:
  deploy:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
      with:
        fetch-depth: 0
    - name: Install mdbook
      run: |
        mkdir mdbook
        curl -sSL https://github.com/rust-lang/mdBook/releases/download/v0.4.14/mdbook-v0.4.14-x86_64-unknown-linux-gnu.tar.gz | tar -xz --directory=./mdbook
        echo `pwd`/mdbook &gt;&gt; $GITHUB_PATH
    - name: Deploy GitHub Pages
      run: |
        # This assumes your book is in the root of your repository.
        # Just add a `cd` here if you need to change to another directory.
        mdbook build
        git worktree add gh-pages gh-pages
        git config user.name "Deploy from CI"
        git config user.email ""
        cd gh-pages
        # Delete the ref to avoid keeping history.
        git update-ref -d refs/heads/gh-pages
        rm -rf *
        mv ../book/* .
        git add .
        git commit -m "Deploy $GITHUB_SHA to gh-pages"
        git push --force
    - name: Deploy Hostgator Pages
      uses: SamKirkland/FTP-Deploy-Action@4.1.0
      with:
        server: ${{ secrets.FTP_HOST }}
        username: ${{ secrets.FTP_USER }}
        password: ${{ secrets.FTP_PASSWORD }}
        server-dir: simplycreate.online/zephyr-guide/
</code></pre>
<p>the problem is FTP-Deploy-Action is syncing the main branch… not the new created files in the previous step in the ?gh-pages branch. what is the best way to sync these files with ftp?</p>

**Comment by ntn888** [comment, marked as answer]
<p>i made a small change add in before: <code>- name: Deploy Hostgator Pages</code></p>
<pre><code class="lang-auto">- name: Checkout gh-pages
      uses: actions/checkout@v2
      with:
        ref: 'gh-pages'
</code></pre>
<p>its’ working now!</p>

**Comment by facundofarias** [comment]
I know you are trying to use a Github Action, but maybe you can give [DeployHQ](https://www.deployhq.com/) a try!
